registry/storage/catalog.go

```diff
@@ -40,22 +41,20 @@ func (reg *registry) Repositories(ctx context.Context, repos []string, last stri
 
 		// if we've filled our array, no need to walk any further
 		if len(foundRepos) == len(repos) {
-			return errFinishedWalk
```

Comment:
This is the way that the one from the Golang stdlib works: https://golang.org/pkg/path/filepath/#Walk
--
I think it's somewhat preferential to move in this direction, because we could eventually get rid of our custom walk function users and just use stdlib for filesystems, and such.

Reply:
It also means that people don't have to do mid-directory interruptions "by hand"